extract_resources_from_api_tree.py

```diff
@@ -0,0 +1,178 @@
+                        logging.info("Reponse code is 200 /successful " + child_resource_endpoint)
+                       
+                        #store the reponse as a dict
+                        response_dict = child_resp.json()
+                        #logging.info("response dict:", response_dict)
+                        
+                        
+                        if node.node_type=='nested':
+                        #store the API JSON response as a dict in the node
+                            node.api_json_list.append(response_dict)
+                            #print(len(node.api_json_list),node.name)
+                            
+
+                            #logging.info(node.api_json_list)
+
+                        elif node.node_type=='normal':
+                            # normalize the dict
+                            temp_df = pd.json_normalize(response_dict['data'])
+
+                            #concatinating the child data frames with child result dataframe
+                            df_child_result = pd.concat([df_child_result, temp_df])
+                        
+                            node.api_response = df_child_result
+
+                        #Logging the child base_url to understand the children which makes the successful response code
+                        logging.info(" The child base_url for the debugging purpose "+ child_resource_endpoint)
+
+                    else:
+
+                        #capture the node names which has end point failure
+                        logging.error("The child resource which not able to make the api call is " + child_resource_endpoint)
+
+                        #Loggingthe node names with API response code
+                        logging.error(node.name + "API response code " + str(child_resp.status_code))
+
+            #Logging the node name as well as node depth for debugging purpose
+             logging.info("The node's depth from its root tree is: " + str(node.name) + "-" + str(node.depth))
+
+            #append the dataframe with child node base_url
+             
+             #list_of_df.append(df_child_result)
+
+
+    #return list_of _df
+    #print(node.node_name)
+    return node.api_response
+
+# Function to build the child base_url to extract the resources from api response
+
+
+def build_child_url(base_url, parent_name, child_name, df_parent):
+
+    """
+    Function traverse the node structure and pick each of the 
+    resources and make the api call and converts the 
+    response into dataframe rows only
+    if the response code is 200(success)
+    :param base_url: base_url
+    :param parent_name: parent node name
+    :param child_name: child node name
+    :param df_parent: dataframe
+    :return:  child-base_url
+    """
+    #intialize a list to store the child_urls
+    child_url_list = []
+
+    if(df_parent is not None and isinstance(df_parent, pd.DataFrame) and not df_parent.empty):
```

Comment:
Check for None and brackets are not needed.

Reply:
Sure will change the logic